service/controller/resource/heartbeatrouting/receiver/receiver.go

```diff
@@ -92,3 +108,16 @@ func getReceiver(cfg alertmanagerconfig.Config, receiver alertmanagerconfig.Rece
 
 	return nil, -1
 }
+
+func cloneHttpConfig(hc *promcommonconfig.HTTPClientConfig) (*promcommonconfig.HTTPClientConfig, error) {
+	s, err := yaml.Marshal(hc)
+	if err != nil {
+		return nil, err
+	}
+	r := &promcommonconfig.HTTPClientConfig{}
+	err = yaml.Unmarshal(s, r)
```

Comment:
That's a good and simple way of doing cloning, nice :)

Reply:
Cool, I'll go ahead with this approach :smile: